pkg/api/v1/resources/resource_interface.go

```diff
@@ -44,16 +45,37 @@ func ProtoCast(res Resource) (ProtoResource, error) {
 
 type InputResource interface {
 	Resource
+	// Deprecated: prefer GetNamespacedStatuses()
 	GetStatus() *core.Status
+	// Deprecated: prefer SetNamespacedStatuses()
 	SetStatus(status *core.Status)
+	GetNamespacedStatuses() *core.NamespacedStatuses
+	SetNamespacedStatuses(namespacedStatuses *core.NamespacedStatuses)
+}
+
+type StatusGetter interface {
```

Comment:
Why are these changes included?

Reply:
> how did you backport...
Lazily, apparently.  Got hit with build issues with a basic cherry-pick, so I ended up reproducing the steps for the original work as a new commit
* update needed dependencies with `go get`
* replace all instances of `option (extproto.equal_all) = true;` with `option (extproto.equal_all) = true;\noption (extproto.clone_all) = true;`
* update `resource_interface.go` with a clean copy from `master` 
* update `pkg/code-generator/cmd/main.go` with a clean copy from `master`